homeassistant/components/nuki/binary_sensor.py

```diff
@@ -0,0 +1,82 @@
+"""Doorsensor Support for the Nuki Lock."""
+
+import logging
+
+from pynuki import STATE_DOORSENSOR_OPENED
+
+from homeassistant.components.binary_sensor import DEVICE_CLASS_DOOR, BinarySensorEntity
+
+from . import NukiEntity
+from .const import (
+    ATTR_LOCK_DOORSENSOR_STATE,
+    ATTR_LOCK_DOORSENSOR_STATE_NAME,
+    ATTR_NUKI_ID,
+    DATA_COORDINATOR,
+    DATA_LOCKS,
+    DOMAIN as NUKI_DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the Nuki lock binary sensor."""
+    data = hass.data[NUKI_DOMAIN][entry.entry_id]
+    coordinator = data[DATA_COORDINATOR]
+
+    entities = []
+
+    for lock in data[DATA_LOCKS]:
```

Comment:
We do not reference between different Entity types, you need to use the nuki_device from library

Reply:
The doorSensor is inside the lock-component in the API (and library). They are not present for the opener devices: https://developer.nuki.io/page/nuki-bridge-http-api-1-12/4#heading--list 
`data[DATA_LOCKS]` contains the `NukiLock` objects from the library and I'm not sure how I should change it here.

Reply:
It's fine for different platforms to access the same library api instance in hass.data. Ie this is not an `Entity` instance.